### DISCUSSION TITLE
How to deel with cycle dependency?

### DISCUSSION THREAD

**Discussion by aquarius-wing** [facebook/lexical#2975] [Q&A] [answered]
Hi, I build a repository just like lexical for learning.
But I got a problem which is "cycle dependency".
This problem cause error like `Super expression must either be null or a function`
or `Cannot access 'xxx' before initialization`.
I don't know what to do, I want to know how you guys deel with this problem?
Thanks!

**Comment by chalecki** [comment, marked as answer]
Check your imports. You can also try some eslint plugins, eg. https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md
